pkg/network/setup/netstat.go

```diff
@@ -90,60 +90,71 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 		sriovIfacesStatusFromDomainHostDevices(domain.Spec.Devices.HostDevices, vmiInterfacesSpecByName)...,
 	)
 
 	var err error
-	interfacesStatus, err = c.updateIfacesStatusFromPodCache(interfacesStatus, vmi.Spec.Domain.Devices.Interfaces, vmi)
+	updatedIfaceStatuses, err = c.updateIfacesStatusFromPodCache(updatedIfaceStatuses, vmi.Spec.Domain.Devices.Interfaces, vmi)
 	if err != nil {
 		return err
 	}
 
 	// Guest Agent information will add and conditionally override data gathered from the cache.
-	interfacesStatus = ifacesStatusFromGuestAgent(interfacesStatus, domain.Status.Interfaces)
+	updatedIfaceStatuses = ifacesStatusFromGuestAgent(updatedIfaceStatuses, domain.Status.Interfaces)
 
-	primaryInterfaceStatus, interfacesStatus := netvmispec.PopInterfaceByNetwork(interfacesStatus, netvmispec.LookupPodNetwork(vmi.Spec.Networks))
+	updatedIfaceStatuses = restoreVirtCtrlOwnedInfo(updatedIfaceStatuses, oldIfaceStatusesByName, vmiInterfacesSpecByName)
+
+	primaryInterfaceStatus, updatedIfaceStatuses := netvmispec.PopInterfaceByNetwork(updatedIfaceStatuses, netvmispec.LookupPodNetwork(vmi.Spec.Networks))
 	if primaryInterfaceStatus != nil {
-		interfacesStatus = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, interfacesStatus...)
+		updatedIfaceStatuses = append([]v1.VirtualMachineInstanceNetworkInterface{*primaryInterfaceStatus}, updatedIfaceStatuses...)
 	}
 
-	interfacesStatus = ifacesStatusFromMultus(interfacesStatus, multusStatusNetworksByName, vmiInterfacesSpecByName)
-
-	vmi.Status.Interfaces = interfacesStatus
+	vmi.Status.Interfaces = updatedIfaceStatuses
 
 	c.removeAbsentIfacesFromVolatileCache(vmi)
 
 	return nil
 }
 
-func ifacesStatusFromMultus(
-	interfacesStatus []v1.VirtualMachineInstanceNetworkInterface,
-	multusStatusNetworksByName map[string]v1.VirtualMachineInstanceNetworkInterface,
+func restoreVirtCtrlOwnedInfo(
+	updatedIfaceStatuses []v1.VirtualMachineInstanceNetworkInterface,
+	oldIfaceStatusesByName map[string]v1.VirtualMachineInstanceNetworkInterface,
 	vmIfacesSpecByName map[string]v1.Interface,
 ) []v1.VirtualMachineInstanceNetworkInterface {
-	for multusIfaceName := range multusStatusNetworksByName {
-		ifaceStatus := netvmispec.LookupInterfaceStatusByName(interfacesStatus, multusIfaceName)
-		_, existInSpec := vmIfacesSpecByName[multusIfaceName]
-		if existInSpec && ifaceStatus == nil {
-			interfacesStatus = append(interfacesStatus, v1.VirtualMachineInstanceNetworkInterface{
-				Name:       multusIfaceName,
-				InfoSource: netvmispec.InfoSourceMultusStatus,
-			})
-		} else if ifaceStatus != nil {
-			ifaceStatus.InfoSource = netvmispec.AddInfoSource(ifaceStatus.InfoSource, netvmispec.InfoSourceMultusStatus)
+	for oldIfaceName, oldIfaceStatus := range oldIfaceStatusesByName {
+		updatedIfaceStatus := netvmispec.LookupInterfaceStatusByName(updatedIfaceStatuses, oldIfaceName)
+
+		ifaceSpec, oldIfaceExistsInSpec := vmIfacesSpecByName[oldIfaceName]
```

Comment:
How about `oldIfaceExistsInSpec` ==> `ifaceSpecExists`. ?

The `if ifaceSpecExists && ifaceSpec.State != v1.InterfaceStateAbsent` reads better and we do not get confused.

Reply:
This change was reverted.